presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/BeginTableWrite.java

```diff
@@ -115,6 +116,23 @@ public PlanNode visitDelete(DeleteNode node, RewriteContext<Context> context)
                     node.getOutputSymbols());
         }
 
+        @Override
+        public PlanNode visitAnalyzeFinish(AnalyzeFinishNode node, RewriteContext<Context> context)
+        {
+            PlanNode child = node.getSource();
+            child.accept(this, context);
```

Comment:
why not `child = child.accept(this, context)` ? if the return from accept() is ignored, why call it all?

also, i'd rename `child` to `source`, but i appreciate it's copy-pasted

Reply:
Nice catch. It should be `child = child.accept(this, context)`